core/src/dird/ua_acl.cc

```diff
@@ -55,7 +55,7 @@ static inline bool is_regex(const char* regexp)
   bool retval = false;
 
   p = regexp;
-  while (p) {
+  while (p && *p) {
```

Comment:
I really don't like this. It is neither clearly understandable what was was broken and how this change fixes it.
I would rather refactor the whole function (it is only a few lines)